analyzer/src/main/kotlin/managers/SpdxDocumentFile.kt

```diff
@@ -122,19 +126,33 @@ internal fun SpdxExternalDocumentReference.getSpdxPackage(packageId: String, wor
 /**
  * Return the [SpdxDocument] this [SpdxExternalDocumentReference]'s [SpdxDocument] refers to.
  */
-private fun SpdxExternalDocumentReference.resolve(workingDir: File): SpdxDocument {
+private fun SpdxExternalDocumentReference.resolve(workingDir: File, issues: MutableList<OrtIssue>): SpdxDocument {
     val uri = runCatching { URI(spdxDocument) }.getOrElse {
         throw IllegalArgumentException("'$spdxDocument' identified by $externalDocumentId is not a valid URI. }")
     }
 
-    if (uri.scheme.equals("file", ignoreCase = true) || !uri.isAbsolute) {
-        val referencedFile = workingDir.resolve(uri.path)
-        val spdxFile = referencedFile.takeIf { it.isFile }
+    var tempDir: File? = null
+
+    val spdxFile = if (uri.scheme.equals("file", ignoreCase = true) || !uri.isAbsolute) {
+        val referencedFile = workingDir.resolve(uri.path).normalize()
+        referencedFile.takeIf { it.isFile }
             ?: throw IllegalArgumentException("The local file URI '$uri' does not point to an existing file.")
-        return SpdxModelMapper.read(spdxFile)
+    } else {
+        tempDir = createOrtTempDir()
+        OkHttpClientHelper.downloadFile(uri.toString(), tempDir).getOrThrow()
+    }
+
+    val hash = Hash.create(checksum.checksumValue, checksum.algorithm.name)
+    val verified = hash.verify(spdxFile)
+    if (!verified) {
+        val referencedFile = tempDir?.let { uri } ?: spdxFile
+        issues += createAndLogIssue(
+            source = externalDocumentId,
```

Comment:
Is `externalDocumentId` actually the identifier of the definition file?

Reply:
Well, no, it's the id of the document referred to.